### PR TITLE
[Workers] Localize a website revision

### PR DESCRIPTION
**Pull request by deadlypants1973**
do not merge

### CODE REVIEW

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 1]
```diff
@@ -295,6 +299,6 @@ theme: dark
 
 ## Resources
 
-In this tutorial, you built and published an i18n tool using `HTMLRewriter`. If you’d like to see the full source code for this application, visit the [repo on GitHub](https://github.com/signalnerve/i18n-example-workers).
+In this tutorial, you built and published an i18n tool using `HTMLRewriter`. If you would like to see the full source code for this application, refer to the [reposotiry on GitHub](https://github.com/signalnerve/i18n-example-workers).
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
In this tutorial, you built and published an i18n tool using `HTMLRewriter`. To review the full source code for this application, refer to the [repository on GitHub](https://github.com/signalnerve/i18n-example-workers).
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 2]
```diff
@@ -155,13 +155,17 @@ class ElementHandler {
 }
 ```
 
-To check that everything looks as expected, use the preview functionality built into Wrangler. Call `wrangler dev` to open up a live preview of your project. `wrangler dev` is refreshed after every code change that you make.
+To review that everything looks as expected, use the preview functionality built into Wrangler. Call [`wrangler dev`]() to open up a live preview of your project. `wrangler dev` is refreshed after every code change that you make.
```
Comment (kodster28) [inline products/workers/src/content/tutorials/localize-a-website/index.md]: Are you missing a link here?

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 3]
```diff
@@ -155,13 +155,17 @@ class ElementHandler {
...
+You can expand on this simple translation functionality to provide country-specific translations, based on the incoming request’s `Accept-Language` header. By taking this header, parsing it, and passing the parsed language into our `ElementHandler`, you can retrieve a translated string in our user’s home language, provided that it’s defined in `strings`.
 
-To implement this, we’ll update the `strings` object, adding a second layer of key-value pairs, and allowing strings to be looked up in the format `strings[country][key]`. In addition, we’ll pass a `countryStrings` object into our `ElementHandler`, so that it can be used during the parsing process. Finally, we’ll grab the `Accept-Language` header from an incoming request, parse it, and pass the parsed language to `ElementHandler`.
+To implement this: 
 
-To parse the `Accept-Language` header, we’ll install the [`accept-language-parser`](https://www.npmjs.com/package/accept-language-parser) NPM package:
+1. Update the `strings` object, adding a second layer of key-value pairs, and allowing strings to be looked up in the format `strings[country][key]`.
```
Comment (kodster28) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
1. Update the `strings` object, adding a second layer of key-value pairs and allowing strings to be looked up in the format `strings[country][key]`. 
```

File: products/workers/src/content/tutorials/localize-a-website/index.md (hunk 1 above)
Comment (kodster28) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
In this tutorial, you built and published an i18n tool using `HTMLRewriter`. If you would like to see the full source code for this application, refer to the [repository on GitHub](https://github.com/signalnerve/i18n-example-workers).
```

**Review by kodster28 — APPROVED** [review]

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 4]
```diff
@@ -38,21 +38,21 @@ theme: dark
 ~/i18n-example $
 ```
 
-The `--site` flag indicates to Wrangler that we want to build a [Workers Sites](/platform/sites) project—this means that there will be both a “site” component, the static HTML that we want to serve to the user, and a Workers script. Inside the Workers script we can customize the HTML response using `HTMLRewriter`.
+The `--site` flag tells Wrangler that you want to build a [Workers Sites](/platform/sites) project. This means that there will be both a site component, the static HTML that you want to serve to the user, and a Workers script. Inside the Workers script, you can customize the HTML response using `HTMLRewriter`.
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
The `--site` flag tells Wrangler that you want to build a [Workers Sites](/platform/sites) project. This means that there will be both a Workers script and a static site component, which includes any HTML and page assets that you want to serve to the user. Inside the Workers script, you can customize the HTML response using `HTMLRewriter`.
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 5]
```diff
@@ -38,21 +38,21 @@ theme: dark
...
 
 The newly generated `i18n-example` project will contain two folders: `public`, which is our static HTML, and `workers-site`:
 
 <pre class="CodeBlock CodeBlock-scrolls-horizontally" language="sh"><code><u><b class="CodeBlock--token-directory">~/i18n-example</b> <b class="CodeBlock--token-prompt">$</b> </u>ls<br/><u><b class="CodeBlock--token-value">public    workers-site  wrangler.toml</b></u></code></pre>
 
-Inside of `public`, we should replace the default generated HTML code with the HTML5 UP template seen in the demo screenshot: you can download a [release](https://github.com/signalnerve/i18n-example-workers/archive/v1.0.zip) (ZIP link) of the code for this project and copy the `public` folder to your own project to get started.
+Inside of `public`, you should replace the default generated HTML code with the HTML5 UP template seen in the demo screenshot: you can download a [release](https://github.com/signalnerve/i18n-example-workers/archive/v1.0.zip) (ZIP link) of the code for this project and copy the `public` folder to your own project to get started.
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
Inside of the `public` directory, replace the default generated HTML code with the HTML5 UP template seen in the demo screenshot: you can download a [release](https://github.com/signalnerve/i18n-example-workers/archive/v1.0.zip) (ZIP link) of the code for this project and copy the `public` folder to your own project to get started.
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 6]
```diff
@@ -38,21 +38,21 @@ theme: dark
...
 
-With the static HTML for this project updated, we can focus on the Workers script inside of the `workers-site` folder, at `index.js`.
+With the static HTML for this project updated, you can focus on the Workers script inside of the `workers-site` folder, at `index.js`.
 
 ## Understanding `data-i18n-key`
 
 The `HTMLRewriter` class provided in the Workers runtime allows developers to parse HTML and write simple JavaScript to query and transform every element of the page.
 
-Our example website is a basic single-page HTML project that lives in `./public`, with some text elements: an `h1` element with the text “Example Site”, and a number of `p` elements with different text:
+The example website in this tutorial is a basic single-page HTML project that lives in `./public`, with different text elements. This includes an `h1` element with the text “Example Site”, and a number of `p` elements with different text:
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
The example website in this tutorial is a basic single-page HTML project that lives in the `public` directory. It includes an `h1` element with the text “Example Site” and a number of `p` elements with different text:
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 7]
```diff
@@ -75,17 +75,17 @@ filename: public/index.html
 </div>
 ```
 
-Using `HTMLRewriter`, we’ll take this page (e.g. `./public/index.html`) and parse the HTML. When we find a `data-i18n-key`, we’ll look up an internal `strings` object, using `data-i18n-key` to find a matching key, and retrieve the string translation. With `HTMLRewriter`, it’s super easy to _query_ elements, for instance, to find a data attribute, but as the name suggests, we can also _rewrite_ elements: taking a translated string and directly inserting it into the HTML.
+Using `HTMLRewriter`, you will take this page (`./public/index.html`) and parse the HTML. When you find a `data-i18n-key`, look up an internal `strings` object, using `data-i18n-key` to find a matching key, and retrieve the string translation. With `HTMLRewriter`, it is easy to query elements to accomplish tasks like finding a data attribute. However, as the name suggests, you can also rewrite elements by taking a translated string and directly inserting it into the HTML.
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
Using `HTMLRewriter`, you will parse the HTML within the `./public/index.html` page. When a `data-i18n-key` attribute is found, you should use the attribute's value to retrieve a matching translation from the `strings` object. With `HTMLRewriter`, it is easy to query elements to accomplish tasks like finding a data attribute. However, as the name suggests, you can also rewrite elements by taking a translated string and directly inserting it into the HTML.
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 8]
```diff
@@ -75,17 +75,17 @@ filename: public/index.html
...
+Begin with the `workers-site/index.js` file. Your Workers application in this tutorial will live entirely in this file.
 
-Inside of this file, the default code for running a [Workers Site](/platform/sites) has been provided. The crucial part of the generated code lives in the `handleEvent` function. The`getAssetFromKV` function retrieves a website asset uploaded from your local `./public` folder, runs some magic to make it live on Workers KV, and returns it to the user. For now, we can ignore much of `getAssetFromKV` (though if you’d like to learn more, check out [the docs](/platform/sites/start-from-worker) .
+Inside of this file, the default code for running a [Workers Site](/platform/sites) has been provided. The crucial part of the generated code lives in the `handleEvent` function. The`getAssetFromKV` function retrieves a website asset uploaded from your local `./public` folder, makes it live on Workers KV, and returns it to the user. For now, ignore `getAssetFromKV` (though if you would like to learn more, refer to [the documentation](/platform/sites/start-from-worker).
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
Inside of this file, the default code for running a [Workers Site](/platform/sites) has been provided. The crucial part of the generated code lives in the `handleEvent` function. The `getAssetFromKV` function retrieves a website asset uploaded from your local `./public` folder, makes it live on Workers KV, and returns it to the user. For now, ignore `getAssetFromKV` (though if you would like to learn more, refer to [the documentation](/platform/sites/start-from-worker).
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 9]
```diff
@@ -75,17 +75,17 @@ filename: public/index.html
...
 
-To implement translations on the site, we’ll take the HTML response retrieved from KV and pass it into a new instance of `HTMLRewriter`. When instantiating `HTMLRewriter`, we can also attach handlers using the `on` function: in our case, we’ll use the `[data-i18n-key]` selector (see the [documentation](/runtime-apis/html-rewriter) for more advanced usage) to parse all elements that require translation with a single class, `ElementHandler`. With the created instance of `HTMLRewriter`, the `transform` function takes a `response` and can be returned to the client:
+To implement translations on the site, take the HTML response retrieved from KV and pass it into a new instance of `HTMLRewriter`. When instantiating `HTMLRewriter`, you can attach handlers using the `on` function. For this tutorial, you will use the `[data-i18n-key]` selector (refer to the [documentation](/runtime-apis/html-rewriter) for more advanced usage) to parse all elements that require translation with a single class, `ElementHandler`. With the created instance of `HTMLRewriter`, the `transform` function takes a `response` and can be returned to the client:
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
To implement translations on the site, take the HTML response retrieved from KV and pass it into a new instance of `HTMLRewriter`. When instantiating `HTMLRewriter`, you can attach handlers using the `on` function. For this tutorial, you will use the `[data-i18n-key]` selector (refer to the [documentation](/runtime-apis/html-rewriter) for more advanced usage) to locate all elements with the `data-i18n-key` attribute, which signifies their need for translation. Any matching element will be passed to an instance of your `ElementHandler` class, which will contain the translation logic. With the created instance of `HTMLRewriter`, the `transform` function takes a `response` and can be returned to the client:
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 10]
```diff
@@ -114,7 +114,7 @@ class ElementHandler {
 }
 ```
 
-With `i18nKey` defined, we can use it to lookup a corresponding translated string. Let’s set up `strings`, an object with key-value pairs corresponding to the `data-i18n-key` value – for now, we’ll define a single example string, `headline`, with a (possibly badly translated) German `string`, “Beispielseite” (“Example Site”), and retrieve it in the `element` function:
+With `i18nKey` defined, you can use it to search for a corresponding translated string. You will now set up `strings`, an object with key-value pairs corresponding to the `data-i18n-key` value. For now, you will define a single example string, `headline`, with a (possibly badly translated) German `string`, “Beispielseite” (“Example Site”), and retrieve it in the `element` function:
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
With `i18nKey` defined, you can use it to search for a corresponding translated string. You will now set up a `strings` object with key-value pairs corresponding to the `data-i18n-key` value. For now, you will define a single example string, `headline`, with a (possibly badly translated) German `string`, “Beispielseite” (“Example Site”), and retrieve it in the `element` function:
```

File: products/workers/src/content/tutorials/localize-a-website/index.md (hunk 2 above)
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]: Missing `wrangler dev` link (if meant for it to be a link)

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 11]
```diff
@@ -249,9 +253,9 @@ async function handleEvent(event) {
 
 ## Publish
 
-Our simple i18n tool built on Cloudflare Workers is complete, and it’s time to deploy it to your domain!
+Your simple i18n tool built on Cloudflare Workers is complete, and it is time to deploy it to your domain.
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
Your simple i18n tool built on Cloudflare Workers is complete and it is time to deploy it to your domain.
```

File: products/workers/src/content/tutorials/localize-a-website/index.md [hunk 12]
```diff
@@ -263,9 +267,9 @@ account_id = "6de123.."
 workers_dot_dev = true
 ```
 
-The `[site]` section at the bottom of `wrangler.toml` tells Wrangler how to deploy your Workers Site. The `bucket` key tells Wrangler where to find your static assets: by default set to the `public` folder, where we placed our HTML code at the beginning of this tutorial. The `entry-point` key indicates where your Workers script is located, and like `bucket`, the default of `workers-site` should already be correctly configured for your application.
+The `[site]` section at the bottom of `wrangler.toml` tells Wrangler how to deploy your Workers Site. The `bucket` key tells Wrangler where to find your static assets: by default set to the `public` folder, where you placed your HTML code at the beginning of this tutorial. The `entry-point` key indicates where your Workers script is located, and like `bucket`, the default of `workers-site` should already be correctly configured for your application.
```
Comment (lukeed) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
The `[site]` section at the bottom of `wrangler.toml` tells Wrangler how to deploy your Workers Site. The `bucket` key tells Wrangler where to find your static assets: by default set to the `public` folder, where you placed your HTML code at the beginning of this tutorial. The `entry-point` key indicates where your Workers script is located and, like `bucket`, the default of `workers-site` should already be correctly configured for your application.
```

File: products/workers/src/content/tutorials/localize-a-website/index.md (hunk 2 above)
Comment (deadlypants1973) [inline products/workers/src/content/tutorials/localize-a-website/index.md]:
```suggestion
To review that everything looks as expected, use the preview functionality built into Wrangler. Call [`wrangler dev`](/cli-wrangler/commands#dev) to open up a live preview of your project. `wrangler dev` is refreshed after every code change that you make.
```

**Pull request merged**